### ISSUE TITLE
Make loss scale public in AMP

### ISSUE THREAD

**Issue by eric-haibin-lin** [apache/incubator-mxnet#17507]
For NLP often users need to do gradient norm clipping. When using AMP, I find it's not easy to access the current gradient scale factor unless I look into AMP source code: https://github.com/dmlc/gluon-nlp/blob/master/scripts/bert/finetune_squad.py#L464 
Shall we make the loss scale a public property? @ptrendx
